src/runtime-rs/crates/resource/src/cdi_devices/container_device.rs

```diff
@@ -0,0 +1,63 @@
+use std::path::Path;
+
+use anyhow::{anyhow, Result};
+use oci_spec::runtime::Spec;
+
+use super::{ContainerDevice, CDI_DEVICE_VENDOR_CLASS};
+use agent::types::Device;
+
+const CDI_PREFIX: &str = "cdi.k8s.io";
+
+fn get_vendor_class<'a>(vendor_id: &'a str, class_id: &'a str) -> Option<&'a str> {
+    let vendor_class = format!("{}-{}", vendor_id, class_id);
+
+    CDI_DEVICE_VENDOR_CLASS.get(&vendor_class[..12]).copied()
```

Comment:
Why extract 0..12 as the key? Please add detail comments.

Reply:
Fixed

Reply:
The `0x030` is not a complete vendor class it should say `0x0300`, `03` means a display controller and then `00` VGA compatible, `01` XGA , `02` 3D and lastly `80` Display controller.

If you have a GPU in compute mode it would show up as 0x0300 and in graphics mode you would see 0x0302. I would suggest to use the complete vendor class, because you may treat graphics vs compute differently..

Reply:
I see what you did down there with the `CDI_DEVICE_VENDOR_CLASS`

Reply:
Ah I should've read the complete code, now it make sense, I think we can ignore the `02` or `00` for the CDI annotation!